### PR TITLE
Remove jQuery class from the commit button

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched from jQuery class functions to plain JavaScript `classList`
- Tested the commit button disabled toggling functionality and it works as before

### CODE REVIEW

File: web_src/js/features/repo-editor.js [hunk 1]
```diff
@@ -148,7 +148,7 @@ export function initRepoEditor() {
       dirtyClass: dirtyFileClass,
       fieldSelector: ':input:not(.commit-form-wrapper :input)',
       change() {
-        const dirty = $(this).hasClass(dirtyFileClass);
+        const dirty = this[0]?.classList.contains(dirtyFileClass);
         commitButton.disabled = !dirty;
       },
```
Comment (silverwind) [inline web_src/js/features/repo-editor.js:153]: Wow, does `areYouSure` really assign `this` to a jQuery collection? That is so not-jquery like. At least the [docs](https://github.com/codedance/jquery.AreYouSure) make me think `this` is a `Element`.

Comment (silverwind) [reply web_src/js/features/repo-editor.js:153]: It does appear to be the case:

 https://github.com/codedance/jquery.AreYouSure/blob/60f199ce6354f9eb8b196575a59610a3c3cebd36/jquery.are-you-sure.js#L130

Can't even prefix `this` with `$`, damnit 😆

**Review by silverwind — APPROVED** [review]

File: web_src/js/features/repo-editor.js (hunk 1 above)
Comment (silverwind) [reply web_src/js/features/repo-editor.js:153]: We can avoid `this` by using first argument `$form`.

File: web_src/js/features/repo-editor.js [hunk 2]
```diff
@@ -148,7 +148,7 @@ export function initRepoEditor() {
       dirtyClass: dirtyFileClass,
       fieldSelector: ':input:not(.commit-form-wrapper :input)',
       change() {
```
Comment (silverwind) [inline web_src/js/features/repo-editor.js]:
```suggestion
      change($form) {
```

File: web_src/js/features/repo-editor.js [hunk 3]
```diff
@@ -147,8 +147,8 @@ export function initRepoEditor() {
       silent: true,
       dirtyClass: dirtyFileClass,
       fieldSelector: ':input:not(.commit-form-wrapper :input)',
-      change() {
-        const dirty = $(this).hasClass(dirtyFileClass);
+      change($form) {
+        const dirty = this[0]?.classList.contains(dirtyFileClass);
```
Comment (silverwind) [inline web_src/js/features/repo-editor.js]:
```suggestion
        const dirty = $form[0]?.classList.contains(dirtyFileClass);
```

File: web_src/js/features/repo-editor.js (hunk 1 above)
Comment (silverwind) [reply web_src/js/features/repo-editor.js:153]: Done in suggestions. @yardenshoham could you test it?

Comment (yardenshoham) [reply web_src/js/features/repo-editor.js:153]: Tested, it works.

**Review by delvh — APPROVED** [review]

**Pull request merged**